### DISCUSSION TITLE
How to get non-english layer names for psd files

### DISCUSSION THREAD

**Discussion by ao-kato1006** [dlemstra/Magick.NET#1702] [Help]
I tried to get the Japanese layer name by referring to discussion ["#620"](https://github.com/dlemstra/Magick.NET/discussions/620#discussioncomment-8943), but in the code below "psdInfo" becomes null. What should I do?
I have confirmed that "layer" itself is working.

```csharp
using ImageMagick;
using ImageMagick.Formats;

string file = "image.psd";

using (MagickImageCollection images = new MagickImageCollection(file)) {
	foreach (MagickImage layer in images) {
		var psdInfo = PsdAdditionalInfo.FromImage(layer);
		if (string.IsNullOrEmpty(psdInfo.LayerName))
			continue;

		Debug.WriteLine(psdInfo.LayerName);
	}
}

**Comment by dlemstra** [comment]
Can you share your file? I suspect your file contains no additional info that contains this information but that it is stored differently.

**Reply by ao-kato1006** [reply]
The file is at the link below.
https://drive.google.com/file/d/1hy-dwu9rkeEbuPqLy96tQMsEogXJ0Qma/view?usp=drive_link

**Comment by dlemstra** [comment]
Attaching files here for future reference:
[SuzukiTsudumi_v1.0.zip](https://github.com/user-attachments/files/16766056/SuzukiTsudumi_v1.0.zip)
[SuzukiTsudumi_v1.0_trm.zip](https://github.com/user-attachments/files/16766057/SuzukiTsudumi_v1.0_trm.zip)

**Comment by dlemstra** [comment]
The first image in the `MagickImageCollection` is a "merged image" that combines data of all the layers and this will not have a layer name. The rest of the layers do have a layer name. For me `psdInfo` is only `null` for the first image. For the rest of the layer I am getting data. You will need to use `string.IsNullOrEmpty(psdInfo?.LayerName)` instead because `psdInfo` can be `null`.

I would advise you to set `<Nullable>enable</Nullable>` in your project.

**Reply by ao-kato1006** [reply]
I was able to solve the problem. Thank you. I just wanted to confirm, is there any documentation that describes the methods of each ImageMagick class? I couldn't find anything, so if there is any, please let me know.

**Reply by dlemstra** [reply]
There is no documentation site for this project. Most methods are discoverable through intellisense in the MagickImage class.

**Reply by ao-kato1006** [reply]
I was trying to use IntelliSense to get the directory that exists in the psd file as advised, but I couldn't find it. How can I get the parent directory name from "MagickImage layer"? This question is different from the Discussions here, so I will open a new Discussion if necessary.

![layer-directory](https://github.com/user-attachments/assets/46a8a3f3-eb62-42a7-b7eb-6afc978e480c)

**Reply by dlemstra** [reply]
Reading those directory names is not possible at the moment. It might be possible but that would probably require quite a lot of research.

**Reply by ao-kato1006** [reply]
I understand. Thank you for your answer.
